src/index.ts

```diff
@@ -96,8 +100,13 @@ export class WaxJS {
     ) => void;
     metricURL?: string;
     returnTempAccounts?: boolean;
+    chainName?: string;
   }) {
-    this.rpc = new JsonRpc(rpcEndpoint);
+    this.rpc = this.registryRpc = new JsonRpc(rpcEndpoint);
+
+    this.chainName = chainName;
+    this.chainId = null;
```

Comment:
no we should have a chainId argument in the constructor so we can set it right away. It can default to null
```
this.chainId = chainId;
```